src/fidesctl/connectors/models.py

```diff
@@ -14,6 +16,32 @@ class AWSConfig(BaseModel):
     aws_access_key_id: str
 
 
+class KeyfileCreds(BaseModel):
+    """The model that holds BigQuery keyfile key/vals"""
+
+    type: Optional[str] = None
+    project_id: str
+    private_key_id: Optional[str] = None
+    private_key: Optional[str] = None
+    client_email: Optional[str] = None
+    client_id: Optional[str] = None
+    auth_uri: Optional[str] = None
+    token_uri: Optional[str] = None
+    auth_provider_x509_cert_url: Optional[str] = None
+    client_x509_cert_url: Optional[str] = None
+
```

Comment:
Is this the standard auth method for all GCP services? Alot of these are optional so I am assuming that this config allows for different authentication methods. Do you foresee that KeyfileCreds could be used for authentication with other GCP services?

Reply:
Can you also envision thing being minified to support the most common auth methods to start with?

Reply:
Yes, I believe this is generic for a service account on GCP based on the [docs I followed](https://cloud.google.com/docs/authentication/production#create_service_account) - while we _could_ have a service account that does everything I wonder if users would prefer separate service accounts for separate functionality potentially? When I set mine up it had pretty restrictive access but I can imagine both scenarios for sure 👍🏽

Reply:
It also doesn't appear to be very practical to reduce the secrets required in my testing this week. So far only a few are able to be stripped away (we still need 7/10)